### DISCUSSION TITLE
help ili9488 & lvgl

### DISCUSSION THREAD

**Discussion by gobbygobby** [arendst/Tasmota#16805] [Q&A]
My issue is I cannot get the lvgl example to display after it succesfully displays the blue tasmota splash screen, it just goes white. what am I missing?
I've used the web installer & suceesfully installed tasmota32-lvgl.bin
I then create a display.ini file & copy over ili9488_XPT2046_display.ini 
I then setup the pins as follows

IO GPIO0	Option A3
IO GPIO5	SPI CS 1
IO GPIO16 XPT2046 CS
IO GPIO17 Display Rst
IO GPIO18 SPI CLK 1
IO GPIO19 SPI MISO 1
IO GPIO21 SPI DC 1
IO GPIO23 SPI MOSI 1

I then either create the autoexec.be or paste the example into the console & get nothing displayed & and the touch has no responce.
here's a copy of the log

00:00:00.003 HDW: ESP32-D0WDQ6 
00:00:00.050 UFS: FlashFS mounted with 304 kB free
00:00:00.091 CFG: Loaded from File, Count 31
00:00:00.100 QPC: Count 3
00:00:00.105 SPI: Hardware using GPIO18(CLK), GPIO23(MOSI) and GPIO19(MISO)
00:00:00.248 BRY: Berry initialized, RAM used=4902 bytes
00:00:00.306 TS: XPT2046
00:00:01.581 RSL: RESULT = {"POWER":"ON"}
00:00:01.583 RSL: POWER = ON
00:00:01.592 LVG: LVGL initialized
00:00:02.110 DSP: Configured display 'ILI9488'
00:00:02.124 Project tasmota - Tasmota Version 12.1.1.5(lvgl-haspmota)-2_0_5(2022-10-12T15:20:36)
00:00:02.929 WIF: Connecting to AP1 SESG Channel 1 BSSId 98:2C:BE:D1:E9:E9 in mode 11n as tasmota-9AE214-0532...
00:00:04.731 WIF: Connected
00:00:04.990 HTP: Web server active on tasmota-9AE214-0532 with IP address 192.168.1.144
10:13:08.637 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.1.1.5(lvgl-haspmota)","FallbackTopic":"cmnd/DVES_9AE214_fb/","GroupTopic":"cmnd/tasmotas/"}}
10:13:08.651 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-9AE214-0532","IPAddress":"192.168.1.144"}}
10:13:09.012 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":18}}
10:13:09.014 RSL: RESULT = {"POWER":"OFF"}
10:13:09.016 RSL: POWER = OFF
10:13:12.244 QPC: Reset

**Comment by s-hadinger** [comment]
Hi, what did you try and what are you trying to achieve?

Seeing the blue splash screen is a good sign

**Reply by gobbygobby** [reply]
I've used lvgl on other projects and know how great it is so would like to use it as a simple remote terminal for home assistant down the shed.
The problem is I cannot get the example code to work. All I'm presented with is a white blank screen to me seems the lvgl lcd interface driver has an issue as I've added some print statements to the example code which appear in the console so prove the script is running 
I've double checked my wiring and temp removed the touch screen wiring and also tried the ili9488 ini without touchscreen support 
All to no joy and have run out of things to try.

**Reply by s-hadinger** [reply]
I'm still not sure what you actually tried. If you don't give more details beyond "it does not work", I can't guess what fails.

You can try a simple example from https://tasmota.github.io/docs/LVGL/

Copy/paste in the Berry console

``` berry
#- start LVGL and init environment -#
lv.start()

hres = lv.get_hor_res()       # should be 320
vres = lv.get_ver_res()       # should be 240

scr = lv.scr_act()            # default screan object
f20 = lv.montserrat_font(20)  # load embedded Montserrat 20

#- Background with a gradient from black #000000 (bottom) to dark blue #0000A0 (top) -#
scr.set_style_bg_color(lv.color(0x0000A0), lv.PART_MAIN | lv.STATE_DEFAULT)
scr.set_style_bg_grad_color(lv.color(0x000000), lv.PART_MAIN | lv.STATE_DEFAULT)
scr.set_style_bg_grad_dir(lv.GRAD_DIR_VER, lv.PART_MAIN | lv.STATE_DEFAULT)

#- Upper state line -#
stat_line = lv.label(scr)
if f20 != nil stat_line.set_style_text_font(f20, lv.PART_MAIN | lv.STATE_DEFAULT) end
stat_line.set_long_mode(lv.LABEL_LONG_SCROLL)                                        # auto scrolling if text does not fit
stat_line.set_width(hres)
stat_line.set_align(lv.TEXT_ALIGN_LEFT)                                              # align text left
stat_line.set_style_bg_color(lv.color(0xD00000), lv.PART_MAIN | lv.STATE_DEFAULT)    # background #000088
stat_line.set_style_bg_opa(lv.OPA_COVER, lv.PART_MAIN | lv.STATE_DEFAULT)            # 100% background opacity
stat_line.set_style_text_color(lv.color(0xFFFFFF), lv.PART_MAIN | lv.STATE_DEFAULT)  # text color #FFFFFF
stat_line.set_text("Tasmota")
stat_line.refr_size()                                                                # new in LVGL8
stat_line.refr_pos()                                                                 # new in LVGL8

#- display wifi strength indicator icon (for professionals ;) -#
wifi_icon = lv_wifi_arcs_icon(stat_line)    # the widget takes care of positioning and driver stuff
clock_icon = lv_clock_icon(stat_line)

#- create a style for the buttons -#
btn_style = lv.style()
btn_style.set_radius(10)                        # radius of rounded corners
btn_style.set_bg_opa(lv.OPA_COVER)              # 100% backgrond opacity
if f20 != nil btn_style.set_text_font(f20) end  # set font to Montserrat 20
btn_style.set_bg_color(lv.color(0x1fa3ec))      # background color #1FA3EC (Tasmota Blue)
btn_style.set_border_color(lv.color(0x0000FF))  # border color #0000FF
btn_style.set_text_color(lv.color(0xFFFFFF))    # text color white #FFFFFF

#- create buttons -#
prev_btn = lv.btn(scr)                            # create button with main screen as parent
prev_btn.set_pos(20,vres-40)                      # position of button
prev_btn.set_size(80, 30)                         # size of button
prev_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)   # style of button
prev_label = lv.label(prev_btn)                   # create a label as sub-object
prev_label.set_text("<")                          # set label text
prev_label.center()

next_btn = lv.btn(scr)                            # right button
next_btn.set_pos(220,vres-40)
next_btn.set_size(80, 30)
next_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)
next_label = lv.label(next_btn)
next_label.set_text(">")
next_label.center()

home_btn = lv.btn(scr)                            # center button
home_btn.set_pos(120,vres-40)
home_btn.set_size(80, 30)
home_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)
home_label = lv.label(home_btn)
home_label.set_text(lv.SYMBOL_OK)                 # set text as Home icon
home_label.center()

#- callback function when a button is pressed, react to EVENT_CLICKED event -#

def btn_clicked_cb(obj, event)
    var btn = "Unknown"
    if   obj == prev_btn  btn = "Prev"
    elif obj == next_btn  btn = "Next"
    elif obj == home_btn  btn = "Home"
    end
    print(btn, "button pressed")
end


prev_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED, 0)
next_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED, 0)
home_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED, 0)
```

**Reply by gobbygobby** [reply]
Sorry that's exactly what I have tried with added print statements to prove the code was running

**Reply by s-hadinger** [reply]
So no error?

What does the splash screen look like? Do you see a Tasmota logo (white on blue background)?

**Reply by gobbygobby** [reply]
Yes that's correct but as soon as it completes drawing, the screen blanks to white as thou it's reset.

**Reply by s-hadinger** [reply]
Can you try `weblog 4` to have more detailed logs?

**Reply by sfromis** [reply]
My experience with a different display (ILI9341) connected is that running that demo (loaded from `autoexec.be`) right after power up results in nothing on the screen - until I turn on the screen using the toggle on the front page (or the equivalent command). It keeps working if I restart, even if the power toggle on the front page now says "OFF". Seems that this "ON" status is remembered outside of what Tasmota is "knowing" about the power state.

**Reply by s-hadinger** [reply]
Oh, good point. The default `DisplayDimmer` value is often too low and the screen looks off. Try `DisplayDimmer 50`

**Reply by sfromis** [reply]
Yeah, quite dim too, but I could still see it - after hitting the power toggle. Of course, next step was exactly dimming to 50 instead of the default 10. And yes, even when the display booted to being off, I still briefly saw the logo splash screen first.

**Reply by gobbygobby** [reply]
[weblog.txt](https://github.com/arendst/Tasmota/files/9783964/weblog.txt)
Thank you guys , 
after I uploaded the log file, I tried sfromis suggestion of trying the power toggle & the display sprang into life.
and now works as it should, maybe worth adding a comment to the docs or changing its initial boot status to prevent others falling into this trap.
